### PR TITLE
PWGHF: Add 3 prong structure with Lc->pKpi in MC ef. monitoring.

### PR DESCRIPTION
**Pull request by mfaggin**
Ciao @fgrosa I create a draft PR just to give you a reference. I tried the code locally and it runs. Here the scripts and config I used: [link](https://cernbox.cern.ch/s/p474LjRDF0it40t)

### CODE REVIEW

**Comment by fgrosa** [comment]
Hi @mfaggin thanks a lot! @jgrosseo could you please have a look if this implementation is fine with you? Thanks!

File: PWGHF/Tasks/taskMcEfficiency.cxx [hunk 1]
```diff
@@ -65,16 +64,24 @@ struct HfTaskMcEfficiency {
...
+    if (std::accumulate(doprocessData.begin(), doprocessData.end(), 0) > 0 && std::accumulate(doprocessMc.begin(), doprocessMc.end(), 0) > 0) {
+      LOGP(fatal, "Data and MC process functions cannot run simultaneously!");
+    }
+
+    auto hCandidates = registry.add<StepTHn>("hCandidates", "Candidate count at different steps", {HistType::kStepTHnF, {axisPt, axisMass, axisPdg, axisCPA, {2, -0.5, 1.5, "collision matched"}, {RecoDecay::OriginType::NonPrompt + 1, RecoDecay::OriginType::None - 0.5, RecoDecay::OriginType::NonPrompt + 0.5}}, kHFNSteps});
+    hCandidates->GetAxis(0)->SetTitle("#it{p}_{T} (GeV/#it{c})");
+    hCandidates->GetAxis(1)->SetTitle("#it{m}_{inv} (GeV/#it{c}^{2})");
     hCandidates->GetAxis(2)->SetTitle("PDG code");
     hCandidates->GetAxis(3)->SetTitle("CPA");
+    hCandidates->GetAxis(5)->SetTitle("origin");
```
Comment (jgrosseo) [inline PWGHF/Tasks/taskMcEfficiency.cxx:79]: Should we call this better? In the end it is prompt or non-prompt, right?

File: PWGHF/Tasks/taskMcEfficiency.cxx [hunk 2]
```diff
@@ -86,23 +93,163 @@ struct HfTaskMcEfficiency {
...
     std::map<int64_t, int> duplicates;
 
-    for (const auto pdgCode : pdgCodes.value) {
+    for (const auto pdgCode : pdgCodes) {
       auto decayType = -1;
       std::array<int, 2> pdgDaughters;
 
       if (pdgCode == pdg::kD0) {
-        decayType = 1 << DecayType::D0ToPiK;
+        decayType = 1 << o2::aod::hf_cand_2prong::DecayType::D0ToPiK;
```
Comment (jgrosseo) [inline PWGHF/Tasks/taskMcEfficiency.cxx:248]:
```suggestion
        decayType = 1 << aod::hf_cand_2prong::DecayType::D0ToPiK;
```

Comment (jgrosseo) [reply PWGHF/Tasks/taskMcEfficiency.cxx:248]: And everywhere else we can skip the `o2::` as we are using the namespace

**Review by jgrosseo — COMMENTED** [review]
Thanks and sorry for my delayed response. I only have a few minor comments...

File: PWGHF/Tasks/taskMcEfficiency.cxx (hunk 1 above)
Comment (mfaggin) [reply PWGHF/Tasks/taskMcEfficiency.cxx:79]: We changed in `Charm hadron origin`, which means: `0 = none`, `1 = prompt`, `2 = non-prompt`.

File: PWGHF/Tasks/taskMcEfficiency.cxx (hunk 2 above)
Comment (mfaggin) [reply PWGHF/Tasks/taskMcEfficiency.cxx:248]: Done, thanks

**Comment by mfaggin** [comment]
Thanks @jgrosseo for your suggestions, I implemented them

**Review by jgrosseo — APPROVED** [review]

**Comment by jgrosseo** [comment]
@TimoWilken Could you check why `macOS-arm` runs O2 tests? Those should be disabled for O2Physics as far as I know... (and they are for the other targets...)

From https://ali-ci.cern.ch/alice-build-logs/AliceO2Group/O2Physics/2275/74087920a30f384fc7d2719ac005f0c293f66228/build_O2Physics_o2_macOS-arm/pretty.html

Unit test results

## sw/BUILD/O2-latest/log
455/485 Test #451: macro/build_geometry.C ...............................................................***Failed    6.61 sec
99% tests passed, 1 tests failed out of 485
100% tests passed, 0 tests failed out of 35

**Review by vkucera — APPROVED** [review]

**Comment by TimoWilken** [comment]
@jgrosseo sorry about that, I've disabled them now.

**Pull request merged**